README.md
# ELS - Enterprise Linux/EuroLinux Sources - opensc
 
## Usage:
  Checkout branch or tag. Use `make sources` to recreate  SOURCE0 archive.


src/ChangeLog
# Generated by Makefile. Do not edit!
commit db60f8da59ef05fa987decfc5a6b5abb9bc93f42
Author: Viktor Tarasov <viktor.tarasov@gmail.com>
Date:   Thu Jun 26 19:15:33 2014 +0200

    release 0.14.0

commit 440289a09113d19d83b20cfecd2ba8dd98ba5826
Author: Andreas Schwier <andreas.schwier@cardcontact.de>
Date:   Tue Jun 17 14:45:45 2014 +0200

    sc-hsm: reduce indicated maximum PIN length to 15
    
    Fix bogus minimum PIN length to support more PIN pad readers

commit cd7fbe3f8c00e9f96dcc5c50a114b9bda9d47215
Author: Raul Metsma <raul@metsma.ee>
Date:   Thu Jun 26 11:49:45 2014 +0300

    MacOSX: Build fat binaries i386...
    
    revert part of 6e255a95
    
    Chrome is still 32 bit only
    This allows plugins running in Chrome (32bit) to use OpenSC PKCS#11, which would otherwise be only 64bit. Tokend "hides" the cpu architecture issue otherwise.

commit 852a4719cfb9b92917bb88cfa4d25bc62dd46374
Author: Viktor Tarasov <viktor.tarasov@gmail.com>
Date:   Mon Jun 9 16:34:06 2014 +0200

    release: OpenSC-0.14.0 RTM

commit 35b74f392379537e1271870a1685f82ee5b5e9f1
Author: Frank Morgner <morgner@informatik.hu-berlin.de>
Date:   Fri Jun 6 22:17:14 2014 +0200

    fixed warning unused variable

commit 359660c45468faa27fb8a5a81d9efcb955538619
Author: Frank Morgner <morgner@informatik.hu-berlin.de>
Date:   Fri Jun 6 23:16:50 2014 +0200

    dnie: removed dead and untested SM wrapping code

commit 55bb8e9ff83b332662dbac202b91230e898d3519
Author: Raul Metsma <raul@metsma.ee>
Date:   Mon Jun 9 12:44:49 2014 +0300

    New generation card-s don't have issues with T1 and 3.5 card with same ATR has issues with T0 (recursive GET BINARY/GET DATA)

commit 072dfeb71cffc053d50a0e5fa0b075335899c6ee
Author: Andreas Schwier <andreas.schwier@cardcontact.de>
Date:   Fri Jun 6 21:10:28 2014 +0200

    sc-hsm-tool: Fixed SEGV if no or invalid card in reader

commit b1b5a39ffad95814901d1955360555d9b3608c5e
Author: Henryk Plötz <henryk@ploetzli.ch>
Date:   Sun May 25 04:49:28 2014 +0200

    Use sc_pkcs15_find_pin_by_auth_id() in asepcos_create_key() to correctly set the newly created key up for use with the configured PIN.

commit 77d8fa390dfff48c5c502c175a0316da57a5c4bc
Author: Nikos Mavrogiannopoulos <nmav@redhat.com>
Date:   Thu Jun 5 11:37:25 2014 +0200

    base64 decoding: Do not assume that char is signed.
    
    In the systems where char is unsigned by default the base64 decoding
    would crash.

commit 2f6b5174a81a7aa72da6efabfd347252e911d41c
Author: Joachim Bauch <bauch@struktur.de>
Date:   Thu Jun 5 11:50:49 2014 +0200

    Support "D-TRUST card 2.4 2ce".

commit 3f13f571c0aa4ecec44ce54f06bab6e7946c2927
Author: Viktor Tarasov <viktor.tarasov@gmail.com>
Date:   Sun Jun 1 19:42:01 2014 +0200

    openpgp-tool: issue-220: read and display OpenPGP data

commit ee0566af090ab89a4446084f98dbcc94fb0be149
Author: Emanuele Pucciarelli <ep@corp.it>
Date:   Sun May 18 00:22:07 2014 +0200

    pkcs11: pr-239: PKCS15 pubkey release fix

commit 77c71be833bbcf10ed807c2320e26efd5b0fe683
Author: Raul Metsma <raul@metsma.ee>
Date:   Wed Apr 30 16:10:27 2014 +0300

    Don't depend configuration default value

commit ccf6da2dbe3e156a68591a7d2b5adfb91a236cae
Author: Raul Metsma <raul@metsma.ee>
Date:   Wed Apr 23 13:01:36 2014 +0300

    Add windows onepin makefile

commit 8e13acf51ebd18d6fa4aeb4fb38c240c11595f3d
Author: Raul Metsma <raul@metsma.ee>
Date:   Wed Apr 23 11:50:38 2014 +0300

    Restore pkcs11 onepin module for Firefox usage

commit f87a364ddcdfe58def14e179c772027f78ce2fb0
Author: Viktor Tarasov <viktor.tarasov@gmail.com>
Date:   Sat May 31 20:02:42 2014 +0200

    release: candidate for the next stable version

commit cfc9461fe5fada9147d018bb434e50df5ea0006a
Author: Viktor Tarasov <viktor.tarasov@gmail.com>
Date:   Fri May 16 16:51:01 2014 +0200

    Update NEWS for opensc-0.13.1

commit 612c7fe43a3d7ffbe85d8e102c582e9f0618786a
Author: Viktor Tarasov <viktor.tarasov@gmail.com>
Date:   Fri May 16 16:50:27 2014 +0200

    git: ignore version.m4.ci

commit 3418e9c1461217b661fc6d5f179d026d08a3cefa
Author: Viktor Tarasov <viktor.tarasov@gmail.com>
Date:   Fri May 16 09:18:17 2014 +0200

    OpenSC 0.13.1 rc1

commit de6d61405b271e22244376e4817e16b49018e1ce
Author: Henrik Andersson <henrik.4e@gmail.com>
Date:   Mon Mar 17 08:44:49 2014 +0100

    Dont use sha256 if openssl is older than 0.9.8.

commit 1df8570a66b401e20d42eeca0752c92fd0eb6e19
Author: Henrik Andersson <henrik.4e@gmail.com>
Date:   Fri Mar 14 09:11:10 2014 +0100

    Add fence against using EVP_sha256 mech.
    
    Which is only available in >=0.9.8

commit e41d94ca1a26230ffe48a26cd796cdf08b7ce870
Author: Viktor Tarasov <viktor.tarasov@gmail.com>
Date:   Tue May 13 14:57:01 2014 +0200

    md: fix return code in 'CardGetContainerInfo'

commit 808fff2246c985c1e69a8aea6bcc71152c5ddbbc
Author: Viktor Tarasov <viktor.tarasov@gmail.com>
Date:   Mon May 12 22:12:29 2014 +0200

    pkcs11: take pubkey CKA_VALUE from pub_data

commit b6ad7a92f58853b0c6be02e29db012d4ef960791
Author: Viktor Tarasov <viktor.tarasov@gmail.com>
Date:   Mon May 12 22:12:29 2014 +0200

    pkcs11: more of debug messages

commit 7dcb4359ffc930c6603dbb656c457669ea213edb
Author: Viktor Tarasov <viktor.tarasov@gmail.com>
Date:   Sun May 11 21:12:38 2014 +0200

    build: put PACKAGE_VERSION_REVISION into version.m4

commit 24fa9d8cc27b54b8929d824563c2d2cc7656ab91
Author: Viktor Tarasov <viktor.tarasov@gmail.com>
Date:   Mon May 5 09:03:28 2014 +0200

    build: by default enable SM support

commit 7b1e2e5dd33fe8d2f3075e927ff198db8afae984
Author: Viktor Tarasov <viktor.tarasov@gmail.com>
Date:   Sun May 11 17:44:05 2014 +0200

    build: uninstall-hook for opensc.conf

commit d4195e297ce244de152cd2dc73a16a984c77e481
Author: Viktor Tarasov <viktor.tarasov@gmail.com>
Date:   Mon May 5 08:46:29 2014 +0200

    build: bootstrap script has expected content
    
    added CI specific 'bootstrap' script version: with parameters it creates the 'm4.version.ci' -- CI specific 'm4.version'


src/win32/winconfig.h
#ifndef _OPENSC_WINCONFIG_H
#define _OPENSC_WINCONFIG_H

#include <stdio.h>
#include <windows.h>
#include <winscard.h>
#include <sys/timeb.h>
#include <sys/stat.h>

#ifdef _MSC_VER 
// TODO fix data truncation instead of disabling them
// VC++ 2015 changes truncation warnings from 4244 to 4267.
#pragma warning (disable : 4267)
#pragma warning (disable : 4244)
#endif

#ifndef strcasecmp
#define strcasecmp stricmp
#endif

#ifndef strncasecmp
#define strncasecmp strnicmp
#endif

#ifndef vsnprintf
#define vsnprintf _vsnprintf
#endif

#ifndef snprintf
#define snprintf _snprintf
#endif

#ifndef R_OK
#define R_OK  4		/* test whether readable.  */
#define W_OK  2		/* test whether writable.  */
#define X_OK  1		/* test whether execubale. */
#define F_OK  0		/* test whether exist.  */
#endif

#ifndef S_IRUSR
#define S_IRUSR S_IREAD
#endif

#ifndef S_IWUSR
#define S_IWUSR S_IWRITE
#endif

#define HAVE_STRNLEN
#define HAVE_IO_H
#define ENABLE_PCSC
#define HAVE_WINSCARD_H
#ifndef DEFAULT_PCSC_PROVIDER
#define DEFAULT_PCSC_PROVIDER "/System/Library/Frameworks/PCSC.framework/PCSC"
#endif

#define ENABLE_NOTIFY

#define SC_PKCS15_PROFILE_DIRECTORY "C:\\Program Files\\OpenSC Project\\OpenSC\\profiles"

#define PATH_MAX FILENAME_MAX

#ifndef PACKAGE_VERSION
#define PACKAGE_VERSION "0.19.0"
#endif

#ifndef VERSION
#define VERSION PACKAGE_VERSION
#endif

#ifndef PACKAGE_NAME
#define PACKAGE_NAME "OpenSC"
#endif

#ifndef OPENSC_FEATURES
#define OPENSC_FEATURES "N/A"
#endif

#ifndef OPENSC_VERSION_MAJOR
#define OPENSC_VERSION_MAJOR 0
#endif

#ifndef OPENSC_VERSION_MINOR
#define OPENSC_VERSION_MINOR 19
#endif

#ifndef OPENSC_VS_FF_COMPANY_NAME
#define OPENSC_VS_FF_COMPANY_NAME  "OpenSC Project"
#endif

#ifndef OPENSC_VS_FF_PRODUCT_NAME
#define OPENSC_VS_FF_PRODUCT_NAME  "OpenSC smartcard framework"
#endif

#ifndef CVCDIR
#define CVCDIR "%PROGRAMFILES%\\OpenSC Project\\OpenSC\\cvc"
#endif

#ifndef DEFAULT_PKCS11_PROVIDER
#define DEFAULT_PKCS11_PROVIDER "${exec_prefix}/lib/opensc-pkcs11.so"
#endif

#ifndef DEFAULT_SM_MODULE
#define DEFAULT_SM_MODULE "libsmm-local.so"
#endif
#ifndef DEFAULT_SM_MODULE_PATH
#define DEFAULT_SM_MODULE_PATH "${exec_prefix}/lib"
#endif

#ifndef OPENSC_SCM_REVISION
#define OPENSC_SCM_REVISION "OpenSC-0.19.0, rev: f1691fc, commit-time: 2018-09-13 13:47:21 +0200"
#endif

#endif
